virttest/qemu_storage.py

```diff
@@ -127,13 +128,24 @@ def create(self, params, ignore_errors=False):
             qemu_img_cmd = qemu_img_cmd.rstrip(",")
 
             if self.base_tag:
+                if params.get("image_json_file.driver"):
+                    jdict = {}
+                    keys = [key for key in params.keys()
+                            if key.startswith('image_json')]
+                    for key in keys:
+                        jdict[key.split("image_json_")[1]] = params[key]
+                    jstring = json.dumps(jdict)
+                    self.base_image_filename = "\'json: {}\'".format(jstring)
                 qemu_img_cmd += " -b %s" % self.base_image_filename
                 if self.base_format:
                     qemu_img_cmd += " -F %s" % self.base_format
 
             qemu_img_cmd += " %s" % self.image_filename
 
             qemu_img_cmd += " %s" % self.size
+        command_prefix = params.get("qemu_img_command_prefix")
```

Comment:
If we add this to init(), may cause other function like info() failed, as they use different mode to execute command in the host. info() call process.run with shell=True.

def _init_subprocess(self):
        if self._popen is None:
            if self.verbose:
                log.info("Running '%s'", self.cmd)
            if self.shell is False:
                cmd = shlex.split(self.cmd)
            else:
                cmd = self.cmd
            try:
                self._popen = subprocess.Popen(cmd,
                                               stdout=subprocess.PIPE,
                                               stderr=subprocess.PIPE,
                                               shell=self.shell,
                                               env=self.env)
            except OSError as details:
                if details.errno == 2:
                    exc = OSError("File '%s' not found" % self.cmd.split()[0])
                    exc.errno = 2
                    raise exc